Move TS plugs and superweapons to the Building queue

pchote opened the pull request:
When the TS mod was first set up, buildings were split between the Building and Support queues following the logic from RA2. Superweapon-related structures (including plugs) went into the support queue, and the power plug naturally followed.

This choice doesn't stand up to scrutiny, and doesn't even match half our official mods (both TD and D2k have superweapon from the Building queue). The current split leads to some weird and inconsistent tech options which I expect would also be a nightmare to balance:
* GDI is able to produce power from either (or both!) the Building and Support queues
* GDI accesses both the Ion Cannon and Hunter Seeker by first committing the Building queue to produce the Upgrade Center, and then the Support queue to produce the two plugs. Nod accesses the Hunter Seeker from the Building queue, and missiles from the Support queue.
* The Waste Refinery is built from the Support queue, despite being a refinery and tech structure with no weapons or abilities of its own.

This PR solves these issues by moving the three GDI plugs and the Nod Stealth Generator, Missile Silo, and Waste refinery to the Building queue. The sidebar order has been updated to give a reasonable unlock flow and keep the same structures in the same place / same hotkey.

Following this new logic the eventual implementation of Concrete will go into the Support queue, the Firestorm Generator on the Building queue, and Firestorm Walls in the Support queue.

This also fixes the Waste Refinery bib, and disables plug production if there aren't any free sockets to place them.

pchote committed:
Fix Waste Refinery bib.

sorcerer86pt commented:
Looks good to me:

GDI:

- Plugs only are buildable if they have a valid place to be put
- No super weapons plug in support but in building tab
- Wall in support tab

Nod:

- Waste Refinery in Building tab
- Missile in Building tab


![Base Profile Screenshot 2020 03 21 - 19 14 15 44](https://user-images.githubusercontent.com/11853415/77234761-f65a1d00-6ba8-11ea-9fc0-4e5083a41349.png)
![Base Profile Screenshot 2020 03 21 - 19 14 36 99](https://user-images.githubusercontent.com/11853415/77234762-f78b4a00-6ba8-11ea-80ec-080d7aefb963.png)
![Base Profile Screenshot 2020 03 21 - 19 17 35 69](https://user-images.githubusercontent.com/11853415/77234763-f823e080-6ba8-11ea-83c4-9c58b70d5215.png)
![Base Profile Screenshot 2020 03 21 - 19 17 42 21](https://user-images.githubusercontent.com/11853415/77234764-f8bc7700-6ba8-11ea-8975-6582ed4691af.png)
![Base Profile Screenshot 2020 03 21 - 19 14 12 65](https://user-images.githubusercontent.com/11853415/77234787-0eca3780-6ba9-11ea-8007-26a12ed32fb8.png)

atlimit8 commented on mods/ts/rules/gdi-support.yaml:
```diff
@@ -137,10 +137,19 @@ GACTWR:
 			tower.sam: !build-incomplete && !tower.vulcan && !tower.rocket && !tower.sam
 	ProvidesPrerequisite@buildingname:
+	ProvidesPrerequisite@pluggable:
+		RequiresCondition: !(tower.vulcan || tower.rocket || tower.sam)
```
I'm on the fence on this one since it would be more compact than distributing the negation if there were more than three condition variables but the negation is distributed above:
```yaml
			tower.vulcan: !build-incomplete && !tower.vulcan && !tower.rocket && !tower.sam
			tower.rocket: !build-incomplete && !tower.rocket && !tower.vulcan && !tower.sam
			tower.sam: !build-incomplete && !tower.vulcan && !tower.rocket && !tower.sam
```
Factoring out the negations above is also an option.

atlimit8 replied:
@abcdefg30 mentioned preferring the distributed form from above on #openra IRC channel.

pchote commented:
Fixed.

pchote committed:
Move plugs and superweapons to Building queue.

pchote committed:
Disable plugs when there are no sockets to place them.

pchote committed:
Rename Defense button tooltip to Support.

pchote commented:
Updated:
* Rewritten (easier than trying to resolve the rebase conflicts) on latest bleed.
* Replaced awkward "(unupgraded)" in plug tooltip text with "socket".
* Renamed "Defense" queue tooltip to "Support" to match TD.

sorcerer86pt commented:
still working 👍 
![Base Profile Screenshot 2020 03 24 - 11 27 11 77](https://user-images.githubusercontent.com/11853415/77421713-347f5880-6dc4-11ea-83da-3ddd6f364ba0.png)
![Base Profile Screenshot 2020 03 24 - 11 27 16 46](https://user-images.githubusercontent.com/11853415/77421718-3517ef00-6dc4-11ea-84da-ede2793d0676.png)
![Base Profile Screenshot 2020 03 24 - 11 27 20 95](https://user-images.githubusercontent.com/11853415/77421720-35b08580-6dc4-11ea-98b6-c0e99d73bd5f.png)
![Base Profile Screenshot 2020 03 24 - 11 27 27 41](https://user-images.githubusercontent.com/11853415/77421721-36491c00-6dc4-11ea-9f58-6a09a18320db.png)
![Base Profile Screenshot 2020 03 24 - 11 29 15 41](https://user-images.githubusercontent.com/11853415/77421723-36e1b280-6dc4-11ea-94b5-de8bcb31ed40.png)
![Base Profile Screenshot 2020 03 24 - 11 29 48 37](https://user-images.githubusercontent.com/11853415/77421724-377a4900-6dc4-11ea-951f-62bbf81d2a69.png)
![Base Profile Screenshot 2020 03 24 - 11 29 58 91](https://user-images.githubusercontent.com/11853415/77421725-3812df80-6dc4-11ea-9d40-66048b468b6e.png)
![Base Profile Screenshot 2020 03 24 - 11 33 02 10](https://user-images.githubusercontent.com/11853415/77421727-38ab7600-6dc4-11ea-9524-b6a142d69b02.png)
![Base Profile Screenshot 2020 03 24 - 11 36 29 09](https://user-images.githubusercontent.com/11853415/77421728-39dca300-6dc4-11ea-9150-796a8b3fc33e.png)
![Base Profile Screenshot 2020 03 24 - 11 38 08 85](https://user-images.githubusercontent.com/11853415/77421731-3b0dd000-6dc4-11ea-8e21-3b17925c9b8b.png)

atlimit8 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/c327090fa16508c3a58603dcbf6d7355c8de5540)
